### PR TITLE
Component/topic form

### PR DESCRIPTION
**Pull request by F-Kirchhoff**
Add Form for adding Topics

### CODE REVIEW

**Review by PReichetanz — APPROVED** [review]
I like it very much!

File: src/app/components/TopicForm/TopicForm.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  onSubmit,
+  onCancel,
+}: TopicFormProps): JSX.Element {
+  const [title, setTitle] = useState('')
+  const [description, setDescription] = useState('')
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault()
+    const newTopic = {
+      id: Math.floor(Math.random() * 10000),
```
Comment (marwinburesch) [inline src/app/components/TopicForm/TopicForm.tsx:24]: I'd recommend a package for that, although you might want to wait implementing one until you get to the backend phase, if you plan to add a backend.
https://github.com/ai/nanoid

File: src/app/components/Dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -81,4 +105,14 @@ const Card = styled.li`
...
+  word-wrap: break-word;
+`
+const FormWrapper = styled.div`
+  z-index: 1000;
+  position: absolute;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  right: 0;
+  background-color: rgba(0, 0, 0, 0.7);
```
Comment (marwinburesch) [inline src/app/components/Dashboard/Dashboard.tsx:117]: This wrapper seems to be on top of the form, I can't click inside the form's inputs

File: src/app/components/Dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -1,41 +1,56 @@
...
 }
 
+type DisplayMsgType = '' | 'SHOW_TOPIC_FORM'
+
 export default function Dashboard({
   content,
   onDisplayToggle,
+  onTopicSubmit,
 }: DashboardProps): JSX.Element {
+  const [displayState, setDisplayState] = useState<DisplayMsgType>('')
```
Comment (marwinburesch) [inline src/app/components/Dashboard/Dashboard.tsx:28]: Do you plan to add other elements to display?

**Review by marwinburesch — COMMENTED** [review]
👍 
A bit big this PR ;)

**Pull request merged**